source/codegen/templates/service_helpers.mako

```diff
@@ -218,8 +218,17 @@ one_of_case_prefix = f'{namespace_prefix}{function_name}Request::{PascalFieldNam
 %>\
 %     if common_helpers.is_struct(parameter) or underlying_param_type == 'ViBoolean':
       std::vector<${underlying_param_type}> ${parameter_name}(${size}, ${underlying_param_type}());
-%     elif service_helpers.is_string_arg(parameter):
+## Byte arrays are leveraging a string as a buffer, so we don't need to take special consideration of the null terminator.
+%     elif service_helpers.is_string_arg(parameter) and parameter['type'] == 'ViInt8[]':
```

Comment:
@zhindes, I just caught this, but `service_helpers.is_string_arg(parameter)` is redundant with `parameter['type'] == `ViInt8[]`, and we should have also included `ViUInt8[]` in the list. Not sure it's worth spinning up a PR to fix it though.

Reply:
Good catches, I'll make a new PR today

Reply:
FYI - We don't seem to use ViUInt8 anywhere, yet. The proto helper only has this case:
```
  if 'ViInt8' in type:
    if is_array == True:
      type = "bytes"
      add_repeated = False
    else:
      type = 'uint32'
```

Reply:
https://github.com/ni/grpc-device/pull/206